services-camel/src/test/java/edu/unc/lib/cdr/ReplicationProcessorIT.java

```diff
@@ -0,0 +1,193 @@
+    }
+
+    @Test
+    public void replicateFileFromFedoraTest() throws Exception {
+        // Create a parent object to put the binary into
+        URI contentBase = createBaseContainer(RepositoryPathConstants.CONTENT_BASE);
+        URI binaryUri = determineRepositoryPath(contentBase);
+
+        String bodyString = "Test text";
+        String filename = "test.txt";
+        String checksum = "82022e1782b92dce5461ee636a6c5bea8509ffee";
+        InputStream contentStream = new ByteArrayInputStream(bodyString.getBytes());
+
+        BinaryObject internalObj = repository.createBinary(binaryUri, "binary_test", contentStream, filename, MIMETYPE, checksum, null);
+
+        when(message.getHeader(CdrBinaryChecksum)).thenReturn(checksum);
+        when(message.getHeader(CdrBinaryUri)).thenReturn(internalObj.getUri().toString());
+
+        processor.process(exchange);
+    }
+    
+    @Test
+    public void replicateExternalFileTest() throws Exception {
+        // Create a parent object to put the binary into
+    	    URI contentBase = createBaseContainer(RepositoryPathConstants.CONTENT_BASE);
+        URI binaryUri = determineRepositoryPath(contentBase);
+
+        String filename = "src/test/resources/external_file.txt";
+        File testFile = new File(filename);
+        InputStream contentStream = new FileInputStream(testFile);
+        String checksum = "9db3fcbaec92b9ccf9aa16f820184813080e77d2";
+
+        BinaryObject externalObj = repository.createBinary(binaryUri, "external_binary_test", contentStream, filename, MIMETYPE, null, null);
+
+        when(message.getHeader(CdrBinaryChecksum)).thenReturn(checksum);
+        when(message.getHeader(CdrBinaryUri)).thenReturn(externalObj.getUri().toString());
+
+        processor.process(exchange);
+    }
+    
+    @Test (expected = ReplicationException.class)
+    public void checksumMismatchTest() throws Exception {
+        // Create a parent object to put the binary into
+    	    URI contentBase = createBaseContainer(RepositoryPathConstants.CONTENT_BASE);
+        URI binaryUri = determineRepositoryPath(contentBase);
+
+        String filename = "src/test/resources/external_file.txt";
+        File testFile = new File(filename);
+        InputStream contentStream = new FileInputStream(testFile);
+        String badChecksum = "41cfe91611de4f56689ca6258237c448d3f91a84";
+
+        BinaryObject externalObj = repository.createBinary(binaryUri, "external_binary_test", contentStream, filename, MIMETYPE, null, null);
+
+        when(message.getHeader(CdrBinaryChecksum)).thenReturn(badChecksum);
+        when(message.getHeader(CdrBinaryUri)).thenReturn(externalObj.getUri().toString());
+
+        processor.process(exchange);
+    }
+    
+    @Test (expected = ReplicationDestinationUnavailableException.class)
+    public void badReplicationLocationTest() throws Exception {
+    	    processor = new ReplicationProcessor(repository, "/some/bad/location", 3, 100L);
```

Comment:
Only this first line is executed since the ReplicationDestinationUnavailableException is immediate.  For clarity it would probably be best to remove the rest of the method.